### PR TITLE
Fixing /IDE/crashers/016-swift-mangle-mangler-mangleidentifier.swift

### PR DESCRIPTION
**Pull request by nkcsgexi**
rdar://problem/29308878
rdar://problem/28387632

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test and merge

**Comment by nkcsgexi** [comment]
@swift-ci Please smoke test and merge

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/AST/Mangle.cpp [hunk 1]
```diff
@@ -1603,7 +1603,12 @@ void Mangler::mangleEntity(const ValueDecl *decl,
       return mangleAccessorEntity(accessorKind, func->getAddressorKind(),
                                   func->getAccessorStorageDecl());
   }
-  
+
+  // Avoid mangling nameless entity. This may happen in erroneous code as code
+  // completion.
+  if (!decl->hasName())
```
Comment (slavapestov) [inline lib/AST/Mangle.cpp]: Why does it not have a name? Would it be better to ensure a name is always set? Or maybe we shouldn't be trying to mangle the entity in the first place? Or perhaps we should fix the mangler to just not allow this?

Comment (nkcsgexi) [inline lib/AST/Mangle.cpp]: This is to improve the mangler to allow nameless entities, in this test case `let:{`. As far as i know, we don't have a default name to set on the var decl lead by `let`, so to fix mangler to be more forgiving is a reasonable solution.

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Pull request merged**